### ISSUE TITLE
testing-library/no-node-access triggered without testing-library imported

### ISSUE THREAD

**Issue by flip-it** [testing-library/eslint-plugin-testing-library#336]
As soon as I access `children` on an arbitrary object, the error `testing-library/no-node-access` is triggered, even when testing-library is not used in this file. It's not even in a test file. So essentially adding a new file with content:

```
const zone = {};
console.log(zone.children);
```

and then running eslint on it fails with message _Avoid direct Node access. Prefer using the methods from Testing Library_.

I have seen the utility [`isTestingLibraryImported`](https://github.com/testing-library/eslint-plugin-testing-library/blob/6e76a2898163c3b7ac8caae18f777d10963ac473/CONTRIBUTING.md#custom-rule-creator) but can't find if it's actually used anywhere. Or am I missing something?

My eslint config is something like:

```
{
    parser: '@babel/eslint-parser',
    plugins: ['ramda', 'jest-dom', 'testing-library'],
    extends: [
        'react-app',
        'airbnb',
        'plugin:ramda/recommended',
        'plugin:jest-dom/recommended',
        'plugin:testing-library/react',
    ]
}
```

**Comment by Belco90** [comment]
Hi @flip-it! I recommend you to read [this thread](https://github.com/testing-library/eslint-plugin-testing-library/issues/331#issuecomment-818070722), so you can find more context about how to lint only your test files and improvements we are gonna apply to clarify this situation.

`isTestingLibraryImported` assumes Testing Library is imported when [the modules mechanism from Aggressive Reporting](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/migrating-to-v4-guide.md#imports) is enabled (which by default is enabled).

Closing this issue in favor of moving any additional discussion to #331
